planner/core/logical_plan_builder.go

```diff
@@ -6711,6 +6735,7 @@ func (b *PlanBuilder) buildCte(ctx context.Context, cte *ast.CommonTableExpressi
 
 // buildRecursiveCTE handles the with clause `with recursive xxx as xx`.
 func (b *PlanBuilder) buildRecursiveCTE(ctx context.Context, cte ast.ResultSetNode) error {
+	b.isCTE = true
```

Comment:
we set true here. where should we set it to false?

Reply:
The default is `false` for non-CTE queries at building. We change this to `true` for recursive and non-recursive CTEs, but this does not affect subsequent queries.